src/bptl/tasks/forms.py

```diff
@@ -71,6 +70,99 @@ def __init__(self, *args, **kwargs):
         self.fields["alias"].required = False
 
 
+class BaseDefaultServiceFormset(BaseInlineFormSet):
+    def clean(self):
```

Comment:
I feel like the clean function has an awful lot of logic in it... Any way to refactor this (later)?

Reply:
I don't really see how - it's a complex mechanism, but it's well tested and contained.